### PR TITLE
Split account_entry from state mod.

### PR DESCRIPTION
**Pull request by kaikai1024**
### Description of the Change

Split account_entry from state mod.

*`remove-gpl` branch needs to rebase it.*

### CODE REVIEW

File: cita-executor/core/src/state/mod.rs [hunk 1]
```diff
@@ -61,106 +63,6 @@ pub struct ApplyOutcome {
...
-            }
-            None => self.account = None,
-        }
-    }
-
-    pub fn account(&self) -> Option<&Account> {
-        self.account.as_ref()
-    }
-}
-
```
Comment (ouwenkg) [inline cita-executor/core/src/state/mod.rs]: Defined again!

File: cita-executor/core/src/state/account_entry.rs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    }
+}
+
+#[derive(Debug, Clone, RlpEncodable, RlpDecodable)]
+pub struct StateProof {
+    address: Address,
+    account_proof: Vec<Bytes>,
+    key: H256,
+    value_proof: Vec<Bytes>,
+}
```
Comment (ouwenkg) [inline cita-executor/core/src/state/account_entry.rs]: The `StateProof` struct duplicated defined here.

Comment (kaikai1024) [reply cita-executor/core/src/state/account_entry.rs]: yes, it's a mistake

File: cita-executor/core/src/state/mod.rs (hunk 1 above)
Comment (kaikai1024) [reply cita-executor/core/src/state/mod.rs]: done

**Pull request merged**